anitya/lib/model.py

```diff
@@ -602,3 +602,129 @@ class ProjectVersion(BASE):
     version = sa.Column(sa.String(50), primary_key=True)
 
     project = sa.orm.relation('Project', backref='versions_obj')
+
+
+class ProjectFlag(BASE):
+    __tablename__ = 'projects_flags'
+
+    id = sa.Column(sa.Integer, primary_key=True)
+
+    project_id = sa.Column(
+        sa.Integer,
+        sa.ForeignKey(
+            "projects.id",
+            ondelete="cascade",
+            onupdate="cascade")
+    )
+
+    project = sa.orm.relation('Project')
+    reason = sa.Column(sa.Text, nullable=False)
+    user = sa.Column(sa.String(200), index=True, nullable=False)
+    state = sa.Column(sa.String(50), default='open', nullable=False)
+
+    created_on = sa.Column(sa.DateTime, default=datetime.datetime.utcnow)
+    updated_on = sa.Column(sa.DateTime, server_default=sa.func.now(),
+                           onupdate=sa.func.current_timestamp())
+
+    def __init__(self, user, project, reason):
+        ''' Constructor.
+        '''
+        self.user = user
+        self.project = project
+        self.reason = reason
+
+    def __repr__(self):
+        return '<ProjectFlag(%s, %s, %s)>' % (self.project.name, self.user,
+                                              self.state)
+
+    def __json__(self, detailed=False):
+        output = dict(
+            id = self.id,
+            project=self.project.name,
+            user=self.user,
+            state=self.state,
+            created_on=time.mktime(self.created_on.timetuple()),
+            updated_on=time.mktime(self.updated_on.timetuple()),
+        )
+        if detailed:
+            output['reason'] = self.reason
+
+        return output
+
+    @classmethod
+    def insert(cls, session, user, project, reason):
```

Comment:
I just added a json representation in case it's used somewhere in fedmsgs or if it would be needed in the future. Should I remove it?